src/pages/profiles/ProfilesPage.tsx

```diff
@@ -0,0 +1,298 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {
+    FunctionComponent,
+    useCallback,
+    useMemo,
+    useRef,
+    useState,
+} from 'react';
+import { FormattedMessage, useIntl } from 'react-intl';
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+    Grid,
+    InputAdornment,
+    Paper,
+    PaperProps,
+    TextField,
+} from '@mui/material';
+import { ManageAccounts } from '@mui/icons-material';
+import {
+    GridButton,
+    GridButtonDelete,
+    GridTable,
+    GridTableRef,
+} from '../../components/Grid';
+import { UserAdminSrv, UserProfile } from '../../services';
+import { useSnackMessage } from '@gridsuite/commons-ui';
+import { Controller, SubmitHandler, useForm } from 'react-hook-form';
+import { GetRowIdParams } from 'ag-grid-community/dist/lib/interfaces/iCallbackParams';
+import { TextFilterParams } from 'ag-grid-community/dist/lib/filter/provided/text/textFilter';
+import { ColDef } from 'ag-grid-community';
+import tickInCircle from 'images/tick-in-circle.png';
+import crossInCircle from 'images/cross-in-circle.png';
+import {
+    RowDoubleClickedEvent,
+    SelectionChangedEvent,
+} from 'ag-grid-community/dist/lib/events';
+import ProfileModificationDialog from './modification/profile-modification-dialog';
+import { UUID } from 'crypto';
+
+const defaultColDef: ColDef<UserProfile> = {
+    editable: false,
+    resizable: true,
+    minWidth: 50,
+    cellRenderer: 'agAnimateSlideCellRenderer', //'agAnimateShowChangeCellRenderer'
+    showDisabledCheckboxes: true,
+    rowDrag: false,
+    sortable: true,
+};
+
+function getRowId(params: GetRowIdParams<UserProfile>): string {
+    return params.data.id;
+}
+
+const ProfilesPage: FunctionComponent = () => {
```

Comment:
Can you split this into several components ?

Reply:
I wonder if a more global refacto with UserPages (and maybe AnnoucementPages) should be done. Separatly ?

Reply:
This one is created in this PR so we can start here. Not a lot, just a ProfilesTable and a AddProfileDialog ?

Reply:
yes done